Marlin/src/pins/ramps/pins_LONGER3D_LKPRO.h

```diff
@@ -0,0 +1,51 @@
+/**
+ * Marlin 3D Printer Firmware
+ * Copyright (c) 2020 MarlinFirmware [https://github.com/MarlinFirmware/Marlin]
+ *
+ * Based on Sprinter and grbl.
+ * Copyright (c) 2011 Camiel Gubbels / Erik van der Zalm
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+#pragma once
+
+// Longer UI assumptions
+#if HOTENDS > 1 || E_STEPPERS > 1
+  #error "Longer UI supports only 1 hotend / E-stepper."
+#endif
+
+#define BOARD_INFO_NAME "Longer-LKPRO"
+
+#ifdef LGT_ESP
+    #define Y_MIN_PIN                         37
+#endif
+
+#define SD_DETECT_PIN                         49
+#define FIL_RUNOUT_PIN                         2
+
+#ifdef U20_Pro_AutoBed
+    #define Z_MIN_PIN                         11
+    #define Z_MAX_PIN                         37
+#else //U30_Pro
+    #define Z_MIN_PIN                         35
+    #ifndef LGT_ESP
+        #define Z_MAX_PIN                     37
+    #endif // !LGT_ESP
```

Comment:
What does `LGT_ESP` mean?

If the pin assignments have to change for the same board in different printers, you should probably have some comments describing what they mean, and why they are not always the same.